src/stapi_fastapi/backends/root_backend.py

```diff
@@ -21,9 +26,38 @@ async def get_order(self, order_id: str, request: Request) -> ResultE[Maybe[Orde
         Should return returns.results.Success[Order] if order is found.
 
         Should return returns.results.Failure[returns.maybe.Nothing] if the order is
-        not found or if access is denied. If there is an Exception associated with attempting to find the order,
-        then resturns.results.Failure[returns.maybe.Some[Exception]] should be returned.
+        not found or if access is denied.
 
-        Typically, a Failure[Nothing] will result in a 404 and Failure[Some[Exception]] will resulting in a 500.
+        A Failure[Exception] will result in a 500.
+        """
+        ...
+
+    async def get_order_statuses(
+        self, order_id: str, request: Request
+    ) -> ResultE[list[OrderStatus]]:
+        """
+        Get statuses for order with `order_id`.
+
+        Should return returns.results.Success[list[OrderStatus]] if order is found.
+
+        Should return returns.results.Failure[Exception] if the order is
+        not found or if access is denied.
+
+        A Failure[Exception] will result in a 500.
+        """
+        ...
+
+    async def set_order_status(
+        self, order_id: str, payload: OrderStatusPayload, request: Request
+    ) -> ResultE[bool]:
```

Comment:
Not really sure what the success return type here should be, so put it as a bool for now.

Reply:
Returning back an OrderStatus makes sense to me